test/unit/test_helpers.py

```diff
@@ -0,0 +1,206 @@
+import copy
```

Comment:
This might be unavoidable, but when I saw the name of this file, I assumed that it was "helpers for the tests", rather than "tests for the _helpers module".

Reply:
I decided to rename `_helpers.py` to the more specific `_all_any.py`, and `test_helpers.py` similarly. If there are other helpers they can go in specifically-named files too.